src/main/java/reposense/model/AuthorConfiguration.java

```diff
@@ -0,0 +1,194 @@
+                && branch.equals(otherAuthorConfig.branch)
+                && authorList.equals(otherAuthorConfig.authorList)
+                && authorEmailsAndAliasesMap.equals(otherAuthorConfig.authorEmailsAndAliasesMap)
+                && authorDisplayNameMap.equals(otherAuthorConfig.authorDisplayNameMap);
+    }
+
+    public Map<Author, String> getAuthorDisplayNameMap() {
+        return authorDisplayNameMap;
+    }
+
+    public void setAuthorDisplayNameMap(Map<Author, String> authorDisplayNameMap) {
+        this.authorDisplayNameMap = authorDisplayNameMap;
+    }
+
+    public String getBranch() {
+        return branch;
+    }
+
+    public void setBranch(String branch) {
+        this.branch = branch;
+    }
+
+    public List<Author> getAuthorList() {
+        return authorList;
+    }
+
+    /**
+     * Sets the details of {@code author} to {@code AuthorConfiguration} including the default alias, aliases
+     * and display name.
+     */
+    private void setAuthorDetails(Author author) {
+        // Set GitHub Id and its corresponding email as default
+        addAuthorEmailsAndAliasesMapEntry(author, Arrays.asList(author.getGitId()));
+
+        addAuthorEmailsAndAliasesMapEntry(author, author.getAuthorAliases());
+        addAuthorEmailsAndAliasesMapEntry(author, author.getEmails());
+
+        setAuthorDisplayName(author, author.getDisplayName());
+    }
+
+    /**
+     * Propagates {@code ignoreGlobList} to {@code author}.
+     */
+    public static void propagateIgnoreGlobList(Author author, List<String> ignoreGlobList) {
+        author.appendIgnoreGlobList(ignoreGlobList);
+    }
+
+    public void addAuthor(Author author) {
+        authorList.add(author);
+        setAuthorDetails(author);
+    }
+
+    public void addAuthor(Author author, List<String> ignoreGlobList) {
+        addAuthor(author);
+        propagateIgnoreGlobList(author, ignoreGlobList);
+    }
+
+    /**
+     * Adds new authors from {@code authorList} and sets the default alias, aliases, emails and display name
+     * of the new authors. Skips authors that have already been added previously.
+     */
+    public void addAuthors(List<Author> authorList, List<String> ignoreGlobList) {
+        for (Author author : authorList) {
+            if (containsAuthor(author)) {
+                logger.warning(String.format(
+                        "Skipping author as %s already in repository %s %s", author.getGitId(), location, branch));
+                continue;
+            }
+
+            addAuthor(author, ignoreGlobList);
+        }
+    }
+
+    public boolean containsAuthor(Author author) {
+        return authorList.contains(author);
+    }
+
+    public void setAuthorList(List<Author> authorList) {
+        this.authorList = authorList;
+    }
+
+    /**
+     * Clears {@code authorEmailsAndAliasesMap} and {@code authorDisplayNameMap} and resets them with the details
+     * of authors in {@code authorList}
```

Comment:
`{code authorEmailsAndAliasesMap}` and `{@code authorDisplayNameMap}` and `{@code authorList}` is not visible to outside classes.
Can just mention that we are updating the author mapping information to its current author list (full stop at the end as well).

Reply:
Updated.